sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java

```diff
@@ -136,19 +142,46 @@ public void addAssignedOverlayPort(long port) {
         return assignedOverlayPorts;
     }
 
-    private static Protos.TaskInfo getTaskInfo(
+    public void setExecutorVolume(VolumeSpec volumeSpec) {
+        // Volumes on the executor must be declared in each TaskInfo.ContainerInfo to be shared among them.
+        for (Protos.TaskInfo.Builder t : getTaskBuilders()) {
+            Protos.ContainerInfo.Builder builder = t.getContainerBuilder();
+
+            if (!builder.hasType()) {
+                builder.setType(Protos.ContainerInfo.Type.MESOS);
+            }
+            builder.addVolumes(getVolume(volumeSpec));
```

Comment:
Could call `getVolume(volumeSpec)` once and pass the resulting `Protos.Volume` to each `TaskInfo`. Not worried about efficiency; it'd just be a bit clearer that it's identical across the TaskInfos.